### PR TITLE
Split functions within the nbnxm setup struct

### PR DESCRIPTION
**Pull request by kanduri**
Dividing the setup tasks based on the GMX data structures they handle

### CODE REVIEW

File: src/gromacs/nblib/gmxsetup.cpp [hunk 1]
```diff
@@ -198,6 +198,31 @@ void NbvSetupUtil::unpackTopologyToGmx()
...
+    // We only use (read) the atom type and charge from mdatoms
+    mdatoms.typeA = const_cast<int*>(system_->topology().getParticleTypeIdOfAllParticles().data());
+    mdatoms.chargeA = const_cast<real*>(system_->topology().getCharges().data());
+    nbv->setAtomProperties(mdatoms, particleInfoAllVdw_);
+}
+
+void NbvSetupUtil::setParticlesOnGrid(std::unique_ptr<nonbonded_verlet_t>& nbv)
+{
+    matrix box_;
+    gmx::fillLegacyMatrix(system_->box().matrix(), box_);
```
Comment (ejjordan) [inline src/gromacs/nblib/gmxsetup.cpp:213]: This is also needed by gmxcalculator, suggesting that this should be computed at a higher level. This is more of a todo suggestion than a change request.

File: src/gromacs/nblib/gmxsetup.h [hunk 2]
```diff
@@ -71,13 +71,27 @@ struct NbvSetupUtil
 {
     NbvSetupUtil(SimulationState system, const NBKernelOptions& options);
 
+    //! Sets up and returns a GmxForceCalculator
+    std::unique_ptr<GmxForceCalculator> setupGmxForceCalculator();
+
+private:
```
Comment (ejjordan) [inline src/gromacs/nblib/gmxsetup.h:78]: If this has private members it should be a class.

File: src/gromacs/nblib/gmxsetup.h [hunk 3]
```diff
@@ -71,13 +71,27 @@ struct NbvSetupUtil
...
 
-    std::unique_ptr<GmxForceCalculator> setupGmxForceCalculator();
+    //! Sets Particle Types and Charges and VdW params
+    void setAtomProperties(std::unique_ptr<nonbonded_verlet_t>& nbv);
 
+    //! Sets up non-bonded verlet on the GmxFOrceCalculator
+    std::unique_ptr<nonbonded_verlet_t> setupNbnxmInstance();
+
+public:
+    //! Pointer to the SimulationState that contains topology, coordinates, velocities, particle/molecule data
```
Comment (ejjordan) [inline src/gromacs/nblib/gmxsetup.h]: We do not need to document what a SimulationState is here.

**Review by ejjordan — CHANGES_REQUESTED** [review]
Mostly ready I would say.

File: src/gromacs/nblib/gmxsetup.h (hunk 3 above)
Comment (kanduri) [reply src/gromacs/nblib/gmxsetup.h]: Done.

File: src/gromacs/nblib/gmxsetup.h (hunk 2 above)
Comment (kanduri) [reply src/gromacs/nblib/gmxsetup.h:78]: Done.

**Review by zhmurov — APPROVED** [review]

**Pull request closed without merging**